server/utilities/distance-utils.js

```diff
@@ -27,10 +27,16 @@ function degToRad(deg) {
  * Validates to ensure that the given lat/lons are valid.
  */
 function validateInputs(lat1, lon1, lat2, lon2) {
+    if (isNaN(lat1) || isNaN(lat2)) {
+        throw new Error("Input is not a number");
+    }
     if (Math.abs(lat1) > 90 || Math.abs(lat2) > 90 || !lat1 || !lat2) {
```

Comment:
I think we should move the !lat1 and !lat2 checks up to to line 30 (and the corresponding change for longitudes).